common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java

```diff
@@ -32,23 +33,46 @@ void tearDown() {
   @Test
   void shouldCheckExistence() {
 
-    byte[] id = new byte[16];
     new Random().nextBytes(id);
-    SubmissionType type = SubmissionType.SUBMISSION_TYPE_PCR_TEST;
 
     assertFalse(repository.exists(id, type.name()));
 
     DiagnosisKey key = DiagnosisKey.builder()
         .withKeyDataAndSubmissionType(id, type)
         .withRollingStartIntervalNumber(600)
         .withTransmissionRiskLevel(2)
+        .withRollingPeriod(1)
         .withCountryCode("DE")
         .withVisitedCountries(Set.of("DE"))
         .withSubmissionTimestamp(0L)
         .withReportType(ReportType.CONFIRMED_TEST)
         .build();
-    service.saveDiagnosisKeys(List.of(key));
+    repository.saveDoNothingOnConflict(
+        key.getKeyData(), key.getRollingStartIntervalNumber(), key.getRollingPeriod(),
+        key.getSubmissionTimestamp(), key.getTransmissionRiskLevel(),
+        key.getOriginCountry(), key.getVisitedCountries().toArray(new String[0]),
+        key.getReportType().name(), key.getDaysSinceOnsetOfSymptoms(),
+        key.isConsentToFederation(), key.getSubmissionType().name());
 
     assertTrue(repository.exists(id, type.name()));
   }
+
+  @Test
+  void shouldNotAddAnInvalidDiagnosisKey() {
+
+    Exception exception = assertThrows(InvalidDiagnosisKeyException.class, () -> {
```

Comment:
id will only contain 0's in this test, is this wanted?

Reply:
Yes, I wanted to check if given a rolling period ourside the range 1-144 -> i have nothing written in the db